test/resolve.source.mjs

```diff
@@ -5,11 +5,19 @@
 
 import fs from 'fs';
 const PKG = JSON.parse(fs.readFileSync('package.json', { encoding: 'utf-8' }));
-export function resolve(specifier, parent, defaultResolve) {
+export function resolve(specifier, context, defaultResolve) {
   if (specifier === PKG.name) {
     specifier = new URL('../tsc-out/temporal.js', import.meta.url).toString();
   } else if (specifier.startsWith('../lib')) {
     specifier = new URL(specifier.replace('../lib/', '../tsc-out/'), import.meta.url).toString();
+    if (!specifier.endsWith('.js')) specifier += '.js';
+  } else if (
+    context.parentURL &&
+    context.parentURL.includes('/tsc-out/') &&
+    specifier.startsWith('./') &&
+    !specifier.endsWith('.js')
+  ) {
+    specifier += '.js';
   }
```

Comment:
Can you document an example of what URIs were being resolved here?

Reply:
(copying from our chat room, sorry for the dupe!) @12wrigja It looks like Node 19 requires a `.js` file extension on all imports, and TSC does not add `.js` to imports as part of transpilation, although babel does. So our demitasse tests (which depend on TSC, unlike our rollup packaging which depends on babel) break on Node 19. There are at least two ways to fix this:

1. Add .js to all our relative imports, like [this commit](https://github.com/js-temporal/temporal-polyfill/pull/203/commits/8bd2c67324da27695e319074d512720857a996b3)
2. Leave our existing polyfill code as-is, and fix the problem in our custom resolver used for demitasse tests, like [this commit](https://github.com/js-temporal/temporal-polyfill/pull/203/commits/15e072e999a6cd9b4a7e0dd617bb30947b1a14a1#diff-b377bd07b17032f760a56c4027320cbcf6401c49f4b4313ae9513f7e74049ee3).

Got a preference?

Reply:
Per our offline convo, I removed (1) and left (2) above.